skyportal/plot.py

```diff
@@ -1629,9 +1654,70 @@ def make_spectrum_layout(obj, spectra, user, device, width):
     )
 
     add_plot_legend(plot, legend_items, width, legend_orientation, legend_loc)
+    # only show this tooltip for spectra, not elemental lines
+    hover.renderers = list(model_dict.values())
+    plot.add_tools(hover)
+
+    smooth_title = Div(text="Smoothing")
+    smooth_input = NumericInput(value=smooth_number)
+    smooth_checkbox = CheckboxGroup(
+        labels=["smoothing"],
+        active=[0] if smoothing else [],
+    )
+    spectrum_plots = [v for k, v in model_dict.items()]
+
+    smooth_callback = CustomJS(
+        args=dict(plots=spectrum_plots, checkbox=smooth_checkbox, window=smooth_input),
+        code="""
```

Comment:
Why not put this in an external file like we do for other CustomJS features?

Reply:
I haven't seen any examples of this in the code I've been working on. Honestly I didn't know that was an option...

Reply:
https://github.com/skyportal/skyportal/blob/master/skyportal/plot.py#L746-L749

Reply:
got it!